packages/synthetic-chain/src/cli/doctor.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { ProposalInfo } from './proposals.ts';
+import assert from 'node:assert';
+import { execSync } from 'node:child_process';
+
+const fixupProposal = (proposal: ProposalInfo) => {
+  const proposalPath = path.join(
+    'proposals',
+    `${proposal.proposalIdentifier}:${proposal.proposalName}`,
+  );
+  console.log('checking proposal', proposalPath, '...');
+  const packageJson = JSON.parse(
+    fs.readFileSync(path.join(proposalPath, 'package.json'), 'utf-8'),
+  );
+  if (packageJson.dependencies) {
+    assert(
+      packageJson.packageManager,
+      'missing packageManager in package.json',
+    );
+    if (packageJson.packageManager.includes('yarn')) {
+      console.log('found "yarn" packageManager, processing...');
+      // ensure it has its own yarn.lock
+      fs.writeFileSync(
+        path.join(proposalPath, 'yarn.lock'),
+        JSON.stringify(packageJson, null, 2),
+      );
+      // verify yarn.lock exists
+      assert(
+        fs.readFileSync(path.join(proposalPath, 'yarn.lock'), 'utf-8'),
+        'missing yarn.lock',
+      );
+      // verify .yarnrc.yml is defined.
+      // TODO verify it's using "nodeLinked: node-modules"
+      assert(
+        fs.readFileSync(path.join(proposalPath, 'yarn.lock'), 'utf-8'),
+        'missing .yarnrc.yml',
+      );
+
+      // refresh install
+      execSync('yarn install', { cwd: proposalPath });
+    }
+  }
+};
+
+export const runDoctor = (proposals: ProposalInfo[]) => {
+  console.log('Running doctor...');
+
+  console.log('enabling corepack');
+  execSync('corepack enable', { stdio: 'inherit' });
+
+  //   path to yarn
+  const yarnpath = execSync('which yarn', {
+    encoding: 'utf-8',
+  });
+  if (yarnpath.includes('homebrew')) {
+    console.error(
+      'Homebrew installs of yarn are not supported. Use corepack instead:',
```

Comment:
I'm curious. How do homebrew yarn installations fail?
Is that worth an issue?

Reply:
something about homebrew providing the wrong version. I got the impression that the claim was that corepack could deal with a requirement for different yarn versions in different repos or different branches.

Reply:
The Homebrew installation prevents Corepack from working